src/mongocxx/write_concern.cpp

```diff
@@ -96,9 +107,25 @@ stdx::optional<std::int32_t> write_concern::nodes() const {
 }
 
 stdx::optional<write_concern::level> write_concern::acknowledge_level() const {
+    stdx::optional<write_concern::level> ack_level;
```

Comment:
I think you can simplify this a bit, since you have the optional, I don't the ternary helsp?

if (w >= 1)
    return stdx::nullopt;

switch (w) {
    case 0:
        return {write_concern::level::k_un...};
    case -2:
        return {write_concern::level::k_default...};

I'm actually not 100% sure that you need the {} around the value. Maybe not? Could you try it and see how it goes? You don't need the break if you are doing an early return.